scipy/spatial/transform/rotation.py

```diff
@@ -281,3 +282,147 @@ def as_rotvec(self):
             return rotvec[0]
         else:
             return rotvec
+
+    @classmethod
+    def _make_elementary_quat(cls, char, angles):
+        if len(char) != 1:
+            raise ValueError("This function computes elementary rotation "
+                             "quaternions about a single axis. Expected "
+                             "argument 1 to be a single character, got "
+                             "{}.".format(char))
+
+        if angles.ndim != 1:
+            raise ValueError("This function computes elementary rotation "
+                             "quaternions about a single axis. Expected "
+                             "argument `angles` to have dimension 1, got "
+                             "{}.".format(angles))
+
+        num_rotations = angles.shape[0]
+        quat = np.zeros((num_rotations, 4))
+
+        char_to_ind = {'x': 0, 'y': 1, 'z': 2}
+        quat[:, 3] = np.cos(angles / 2)
+        quat[:, char_to_ind[char]] = np.sin(angles / 2)
+        return quat
+
+    @classmethod
+    def _compose_quat(cls, p, q):
+        # p and q should have same shape (N, 4)
+        if p.shape != q.shape:
+            raise ValueError("Expected same shape for input quaternions, "
+                             "got {0} and {1}".format(p.shape, q.shape))
+
+        num_quat = p.shape[0]
+        product = np.empty((num_quat, 4))
+        product[:, 0] = (p[:, 3] * q[:, 0] + p[:, 0] * q[:, 3] +
+                         p[:, 1] * q[:, 2] + p[:, 2] * q[:, 1])
+        product[:, 1] = (p[:, 3] * q[:, 1] - p[:, 0] * q[:, 2] +
+                         p[:, 1] * q[:, 3] + p[:, 2] * q[:, 0])
+        product[:, 2] = (p[:, 3] * q[:, 2] + p[:, 0] * q[:, 1] -
+                         p[:, 1] * q[:, 0] + p[:, 2] * q[:, 3])
+        product[:, 3] = (p[:, 3] * q[:, 3] - p[:, 0] * q[:, 0] -
+                         p[:, 1] * q[:, 1] + p[:, 2] * q[:, 2])
+
+        return product
+
+    @classmethod
+    def _elementary_quat_compose(cls, seq, angles, intrinsic=False):
+        seq = seq.lower()
+        num_rotations = angles.shape[0]
+        quat = np.empty((num_rotations, 4, len(seq)))
```

Comment:
If I'm not mistaken this array is not necessary. You can generate elementary quaternions on the fly.